ast/src/parsed/asm.rs

```diff
@@ -362,11 +351,9 @@ impl AbsoluteSymbolPath {
         Self { parts }
     }
 
-    /// Formats the path without leading `::` and uses `.` as separator if
-    /// there are at most two components.
-    pub fn to_dotted_string(&self) -> String {
-        let separator = if self.parts.len() <= 2 { "." } else { "::" };
-        self.parts.join(separator)
+    // Returns the path as a string without the leading "::".
+    pub fn to_relative_string(&self) -> String {
```

Comment:
I'm not 100% convinced by this. What is the string relative to?
In some parts of the cod, we use `.relative_to(&Default::default())` and then format the result. This is the same as this function.

What if we implement `.relative_to_root()`, which is much simpler than `relative_to`. It would return a SymbolPath which can then be formatted. If you implement that, you should also change all other places where we use `relative_to(&Default::default())`.